pants.ini

```diff
@@ -302,9 +301,6 @@ skip: True
 [pycheck-newstyle-classes]
 skip: True
 
-[mypy]
-config_file: build-support/mypy/mypy.ini
```

Comment:
See above about reverting this.

Reply:
I totally agree, a lot of these changes were to subvert a failing lint shard in ci, but this way I don't think mypy works as intended anymore.

Reply:
Mypy isn't used much in pants, so I think we're sticking with this for now. We should eventually find a way to avoid the c++ resolving issue(s).

Reply:
I think that we do actually need to configure this right now.

If we were going to temporarily stop configuring it, the right way to do that would be with a `TODO` here that disables it and points to a ticket for re-enabling it. But as Eric said, I don't think we should do that without more discussion.